src/headless-cms/apollo/__tests__/apolloClient.test.ts

```diff
@@ -0,0 +1,104 @@
+import { ApolloClient } from '@apollo/client';
+
+import { initializeCMSApolloClient, resetApolloClient } from '../apolloClient';
+
+describe('initializeCMSApolloClient', () => {
+  const initialState = {
+    ROOT_QUERY: {
+      __typename: 'Query',
+      testField: 'testValue',
+    },
+  };
+
+  beforeEach(() => {
+    resetApolloClient();
+  });
+
+  it('should create a new Apollo Client instance if no instance exists', () => {
+    const client = initializeCMSApolloClient();
+    expect(client).toBeInstanceOf(ApolloClient);
+  });
+
+  it('should return the existing Apollo Client instance if one exists', () => {
+    const existingClient = initializeCMSApolloClient();
+    const client = initializeCMSApolloClient();
+    expect(client).toBe(existingClient);
+    expect(client).toBeInstanceOf(ApolloClient);
+  });
+
+  it('should hydrate the cache with initial state if provided', () => {
+    const client = initializeCMSApolloClient();
+    const restoreSpy = vitest.spyOn(client.cache, 'restore');
+    const extractSpy = vitest.spyOn(client, 'extract');
+
+    initializeCMSApolloClient(initialState);
+
+    expect(extractSpy).toHaveBeenCalled();
+    expect(restoreSpy).toHaveBeenCalledWith(
+      expect.objectContaining(initialState)
+    );
+  });
+
+  it('should restore the cache with merged existing and initial states', () => {
```

Comment:
Same comment here as in https://github.com/City-of-Helsinki/palvelutarjotin-admin/pull/378/files#r2005059243